pkg/crypto/hmac/hmac.go

```diff
@@ -40,3 +42,23 @@ func encode(l *lua.State, h func() hash.Hash) int {
 	l.PushString(string(mac.Sum(nil)))
 	return 1
 }
+
+func signsha256_multi(l *lua.State) int {
+	return encode_multi(l, sha256.New)
+}
+
+func encode_multi(l *lua.State, h func() hash.Hash) int {
+	messages, err := util.PullStringArray(l, 1)
+	if err != nil {
+		lua.Errorf(l, err.Error())
+		panic("unreachable")
+	}
+	key := lua.CheckString(l, 2)
+
+	mac := hmac.New(h, []byte(key))
+	for _, message := range messages {
+		mac.Write([]byte(message))
```

Comment:
This looks to do what `signsha256` does over an array of strings instead of just 1 string. Is there a reason we can't just do iterative `signsha256` calls in the code instead of adding this helper?

Reply:
What would that look like? The result of the first call being the key for the second call? If so, I don't think that would work as this fails:
```lua
equals(
  "successive calls to signsha256 are the same as a single call to signsha256_multi",
  hmac.signsha256("bar", hmac.signsha256("foo", "secret")),
  hmac.signsha256_multi(array({ "foo", "bar" }), "secret")
)
```

Reply:
Is the requirement that the key changes over the course of the encryption of messages? I was thinking if the key remains the same, the messages could be called 1 by 1. For example, instead of doing 

```
hmac.signsha256_multi(array({"foo", "bar}), "key")
```

We could do 

``` 
hmac.signsha256("foo","key")
hmac.signsha256("bar","key")
```

Reply:
> Is the requirement that the key changes over the course of the encryption of messages?

No, the key stays the same and we add multiple messages. Equivalent to the [`OpenSSL::HMAC#update` method](https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/HMAC.html#method-i-update)

> We could do
> 
> ```
> hmac.signsha256("foo","key")
> hmac.signsha256("bar","key")
> ```

How would we get a single value out of that?